js/src/sheet.ts

```diff
@@ -253,56 +255,35 @@ Handsontable.renderers.registerRenderer('styled', function customRenderer(hotIns
     });
 });
 
-let testing = false;
-let setTesting = function() {
-    testing = true;
-};
-
 let SheetView = widgets.DOMWidgetView.extend({
     render: function() {
-        this._refresh_requested = false;
-        /*
-        We debounce rendering of the table, since rendering can take quite some time, however that
-        makes unittesting difficult, since the results don't happend instanely. Maybe a better solution
-        is to use a deferred object.
-        */
-        if(testing) {
-            this.throttled_on_data_change = this._real_on_data_change;
-            this.throttled_render = this._real_table_render;
-
-        } else {
-            this.throttled_on_data_change = debounce(() => this._real_on_data_change(), 100);
-            this.throttled_render = debounce(() => this._real_table_render(), 100);
-        }
-        //
-        //this.listenTo(this.model, 'change:data', this.on_data_change)
         this.displayed.then(() => {
-            this._build_table().then(hot => {
+            this._build_table().then((hot) => {
                 this.hot = hot;
-                Handsontable.hooks.add('afterChange',    () => this._on_change(),      this.hot);
```

Comment:
The fact that we were calling `() => this._on_change()` was causing an infinite loop of events, because parameters for `_on_change` were not given and this part of the code were not fulfilling its purpose:
```javascript
_on_change: function(changes, source) {
    if(source == 'loadData')
        return; // ignore loadData
...
```
Meaning that we had a loop like: `loadData(...)` -> `_on_change` -> `this.model.set('data', ...)` -> `on_data_change(...)` -> `loadData(...)` ...

I don't know why it was even working. Maybe it was just not complaining about the infinite loop and was just running indefinitely and asynchronously.

Reply:
Actually, this is due to the fact that I'm always explicitly triggering the `change:data` event, I need to dig more into it

Reply:
Yes, I'm afraid it wasn't trivial :) Hope you solve it!

Reply:
Btw, I originally had this 'Lock' object that we moved a few days ago to avoid this infinite loop, in case you need it.

Reply:
Oh ok thanks, that might help

Reply:
```javascript
if(source == 'loadData')
        return; // ignore loadData
```
solves this infinite loop thing, but it introduces a bug: the data is not sent to the Python side anymore if there is a change on the JavaScript side.

Reply:
Solved